Sources/PackageSigning/Signing.swift

```diff
@@ -0,0 +1,85 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the Swift open source project
+//
+// Copyright (c) 2023 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See http://swift.org/LICENSE.txt for license information
+// See http://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+//
+//===----------------------------------------------------------------------===//
+
+import Dispatch
+import Foundation
+
+import Basics
+
+public struct Signer {
+    public init() {
+    }
+
+    public func sign(
+        _ content: Data,
+        with identity: SigningIdentity,
+        in format: SignatureFormat,
+        observabilityScope: ObservabilityScope,
+        callbackQueue: DispatchQueue,
+        completion: @escaping (Result<Data, Error>) -> Void
```

Comment:
Do we want callback or async/await?

Reply:
Don't have strong opinions, but if we want/need this to work on macOS CI, we cannot use async/await.

Reply:
starting to move things to async/await is better. the CI issues will wash away sooner or later

Reply:
cc @shahmishal for the CI issues

Reply:
Hard for me to tell, we may actually need newer macOS. It depends on whether Xcode offering the concurrency backwards compatibility library will be sufficient or if it has to come from the superior toolchain.

Reply:
👍 changing to async/await